ng-dev/utils/git/authenticated-git-client.ts

```diff
@@ -64,6 +68,12 @@ export class AuthenticatedGitClient extends GitClient {
    * provided OAuth scopes.
    */
   async hasOauthScopes(testFn: OAuthScopeTestFunction): Promise<true | {error: string}> {
+    // Because bot accounts do not have the same structure for OAuth scopes, we always assume they
+    // have the correct access.
+    if (this.userType === 'bot') {
+      return true;
+    }
```

Comment:
We have to be explicitly told if the account is a bot or a token, we can't determine it from API calls.  Additionally, because the token for an App is not an oauth token, it doesn't have the oauth headers we look for here.

Reply:
Okay, that makes sense. I think always returning true here is a bad practice though. How about:

* Throwing for bots in this helper here? Introducing a new unique error class for such unsupported api calls with a bot token
* Manually handling the bot scenario in the merge tool before calling this method

Reply:
I handled it in the merge tool, with a todo on how to properly assess it.